### PR TITLE
direct stderr output to stderr again

### PR DESCRIPTION
**Pull request by rhaschke**
Fix for #396.

### CODE REVIEW

File: catkin_tools/verbs/catkin_build/cli.py [hunk 1]
```diff
@@ -158,6 +158,8 @@ def prepare_arguments(parser):
         help='Print output from commands in ordered blocks once the command finishes.')
     add('--interleave-output', '-i', action='store_true', default=False,
         help='Prevents ordering of command output when multiple commands are running at the same time.')
+    add('--split-stderr', '-e', action='store_false', dest='merge_stderr_into_stdout', default=True,
+        help='Direct stderr output to stderr again, instead of merging it with stdout (which is the default)')
```
Comment (NikolausDemmel) [inline catkin_tools/verbs/catkin_build/cli.py]: The help should maybe not contain the "again"?

**Comment by NikolausDemmel** [comment]
I notice that when running `catkin build -v -j1` I get the output from stdout not continuously but all in one go once it is finished (for each build stage). I'm not sure if that is supposed to be that way or not, but in any case, how does that play together with this addition.

Would I get the full output of stderr and stdout interleaved correctly, or would I get them as two blocks one after another and not interleaved and I run with the added option to split stdout and stderr?

**Comment by rhaschke** [comment]
> I notice that when running `catkin build -v -j1` I get the output from stdout not continuously.

I guess, this is intended. Otherwise output from different jobs might be interleaved. You can enforce interleaved output with option `--interleave-output`. I'm not sure, catkin automatically evaluates `-j1` to modify it's output mode.

> Would I get the full output of stderr and stdout interleaved correctly, or would I get them as two blocks one after another and not interleaved when I run with the added option to split stdout and stderr?

Didn't tried this, yet. In any case, the actual output you will see strongly depends on the buffering mode that is performed on stdout/stderr. By default stdout is buffered, while stderr is not, which will mix up the actual output!

**Comment by wjwwood** [comment]
@NikolausDemmel You would need to use `-i` if you wanted the output from each build command to "stream" to the screen, otherwise it will just print in blocks as each step finishes.

**Comment by NikolausDemmel** [comment]
Thanks for the hints. I wasn't actually aware of that flag.

**Comment by wjwwood** [comment]
Oh and you don't need the `-i` if you use `-p1`. Basically, `-p1` ensures only one process would be printing at a time and therefore it is safe to print immediately without a prefix. The `-i` is designed for when you have multiple processes in flight and you need to prefix them with the package name (and stage I think) so that you can disambiguate what came from where when they interleave.

**Comment by wjwwood** [comment]
Looking at this now, is there any benefit to outputting the error to stdout? It is already being captured separately, this only affects the printing, and so it shouldn't affect the output ordering. A calling program could always pipe the stderr of catkin build to stdout if they wished. I'm trying to come up with a reason that catkin build should need to do it itself.

@jbohren I imagine you're super busy at your new gig (congrats btw 😄), but perhaps you could give your perspective on why errors currently go to stdout in the first place?

I'm inclined to drop the option and just output errors to stderr always.

**Comment by wjwwood** [comment]
I created https://github.com/catkin/catkin_tools/pull/424 to replace this one (could not push here), but I kept the commits from this pr. Thanks for pushing the initial pr @rhaschke.

**Pull request closed without merging**